models/opensea/ethereum/opensea_v4_ethereum_trades.sql

```diff
@@ -8,62 +15,525 @@
-      ,is_private
-      ,'seaport-' || CAST(tx_hash AS VARCHAR(100)) || '-' || cast(evt_index as VARCHAR(10)) || '-' || CAST(nft_contract_address AS VARCHAR(100)) || '-' || cast(token_id as VARCHAR(100)) || '-' || cast(sub_idx as VARCHAR(10)) as unique_trade_id
-  from {{ ref('seaport_ethereum_trades') }}
- where CAST(zone_address AS VARCHAR(100)) in ('0xf397619df7bfd4d1657ea9bdd9df7ff888731a11'
-                       ,'0x9b814233894cd227f561b78cc65891aa55c62ad2'
-                       ,'0x004c00500000ad104d7dbd00e3ae0a5c00560c00'
-                       ,'0x110b2b128a9ed1be5ef3232d8e4e41640df5c2cd'
-                       ,'0x000000e7ec00e7b300774b00001314b8610022b8' -- newly added on Seaport v1.4
-                       )
-   and version = 'v2'                       
+{% set c_seaport_first_date = "2023-02-01" %}
+{% set c_native_token_address = "0x0000000000000000000000000000000000000000" %}
+{% set c_alternative_token_address = "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2" %}
+{% set c_native_symbol = "ETH" %}
+{% set c_seaport_contract_address = "0x00000000000001ad428e4906ae43d8f9852d0dd6" %} -- v2 = Seaport v1.4
+
+with source_ethereum_transactions as (
+    select *
+    from {{ source('ethereum','transactions') }}
+    {% if not is_incremental() %}
+    where block_time >= date '{{c_seaport_first_date}}'  -- seaport first txn
+    {% endif %}
+    {% if is_incremental() %}
+    where block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+)
+,ref_tokens_nft as (
+    select *
+    from {{ ref('tokens_nft') }}
+    where blockchain = 'ethereum'
+)
+,ref_tokens_erc20 as (
+    select *
+    from {{ ref('tokens_erc20') }}
+    where blockchain = 'ethereum'
+)
+,ref_nft_aggregators as (
+    select *
+    from {{ ref('nft_aggregators') }}
+    where blockchain = 'ethereum'
+)
+,ref_nft_aggregators_marks as (
+    select *
+    from {{ ref('nft_ethereum_aggregators_markers') }}
+)
+,source_prices_usd as (
+    select *
+    from {{ source('prices', 'usd') }}
+    where blockchain = 'ethereum'
+    {% if not is_incremental() %}
+      and minute >= date '{{c_seaport_first_date}}'  -- seaport first txn
+    {% endif %}
+    {% if is_incremental() %}
+      and minute >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+)
+,iv_orders_matched AS (
+    select evt_block_time as om_block_time
+          ,evt_tx_hash as om_tx_hash
+          ,evt_index as om_evt_index
+          ,posexplode(orderhashes) as (om_order_id, om_order_hash)
+      from {{ source('seaport_ethereum','Seaport_evt_OrdersMatched') }}
+     where contract_address = '{{c_seaport_contract_address}}'
+)
+,iv_platform_fee_wallet (wallet_address, wallet_name) as (
+    values   ('0x5b3256965e7c3cf26e11fcaf296dfc8807c01073','opensea')
+            ,('0x8de9c5a032463c561423387a9648c5c7bcc5bc90','opensea')
+            ,('0x34ba0f2379bf9b81d09f7259892e26a8b0885095','opensea')
+            ,('0x0000a26b00c1f0df003000390027140000faa719','opensea')
+)
```

Comment:
One thing to keep in mind here is that there are other platforms that are using Seaport.  Blur, Gem, and Origin come to mind.  I'm not sure if they're all doing platform fees and I don't have them all available but would you be interested in them when I come across them?

Reply:
If some transfer destination is in that addresses, then it would be considered as a `platform_fee`. If not in this list, it would be considered as `royalty_fee` (or creator_earning). As far as I know, there are ens.vision and alphasharks that receive fees.
FYI, this is a correction for opensea_ethereum.trades and nft.trades. Does not affect seaport_ethereum.trades directly. But I'm trying to fix it in this way sooner or later.

Reply:
oh, so it is.  The issue I reported was in `seaport_ethereum_base_pairs`.  That looks like the dep that handles the `is_creator_fee` logic this depends on.  Not sure where it makes the most sense, but it would be cool if it wasn't only supporting of opensea platform fees.  Your call, thanks for working on this.

Reply:
Since seaport has no notion of creator/marketplace fees, only payments, we should aim to move any creator fee logic out of the seaport models.
That should be handled by the marketplace specific models which build on top of the seaport ones. Not something we want to do in this PR, but definitely something we should do in the future to improve further.

Reply:
I get where you're coming from.  However, a bit of a snag on that is that there's no clean way to separate all the different marketplaces out on Seaport.  Just recently it's become the norm to append a domain hash to the txdata, but it's not enforced and isn't on historical transactions.  

That said, I'm not offering up any grand ideas here, it's just to point out that this spellbook isn't filtering out other marketplaces and may count other platform fees on Seaport as creator royalties.

Reply:
I'm not exactly sure I understand the concept of Seaport, but I would like to tell you what I think.

- Seaport was developed by OpenSea, but it was developed as an open source platform that anyone can use. That's why the schema name is Seaport not OpenSea, and it's actually freely used by several marketplaces. 

- However, as 0xRob mentioned, there is no tagging like the purpose of the token being transferred (creator earnings/platform fees). I initially thought it was an order, but I was wrong. I guess it's because it's a flexible platform for various trades.

- This PR is for `opensea.trades`. I wanted to modify Seaport to be universally usable if possible and just filter out OpenSea trades, but that proved to be difficult, so I ended up specifying specific addresses like this PR. I don't think this logic is perfect either, I just think it's the best I could do.

- It's not clear how to distinguish between each marketplace in Seaport. I thought maybe `zone_address` was a separate column for this purpose, but now other marketplaces are using the same value as OpenSea, so it's quite difficult to find transactions from OpenSea in Seaport. For reference, the domain hash is used as a value to determine where the transaction came from. Currently, OpenSea trades are determined by `fee_receive_address` and `zone_address`.